sdk/common/src/main/java/io/opentelemetry/sdk/resources/ResourcesConfig.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.resources;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.sdk.common.export.ConfigBuilder;
+import io.opentelemetry.trace.Span;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Class that holds global resources parameters.
+ *
+ * <p>Configuration options for {@link ResourcesConfig} can be read from system properties,
+ * environment variables, or {@link Properties} objects.
+ *
+ * <p>For system properties and {@link Properties} objects, {@link ResourcesConfig} will look for
+ * the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.config.resources.providers.disable}: to set the ResourceProvider service
+ *       providers found on the classpath to be disabled.
+ * </ul>
+ *
+ * <p>For environment variables, {@link ResourcesConfig} will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code OTEL_CONFIG_RESOURCES_PROVIDERS_DISABLE}: to set the ResourceProvider service
```

Comment:
I would change this name to `OTEL_JAVA_DISABLE_RESOURCES_PROVIDES`. According to the spec, this is a language-specific feature which requires the convention `OTEL_{LANGUAGE}_{FEATURE}`: [Spec link](https://github.com/open-telemetry/opentelemetry-specification/blob/14b5b6a944e390e368dd2e2ef234d220d8287d19/specification/sdk-environment-variables.md#language-specific-environment-variables)